mlflow/artifacts/__init__.py

```diff
@@ -57,3 +59,107 @@ def download_artifacts(
     artifact_repo = get_artifact_repository(artifact_uri)
     artifact_location = artifact_repo.download_artifacts(artifact_path, dst_path=dst_path)
     return artifact_location
+
+
+def load_text(artifact_uri: str) -> str:
+    """
+    Loads content of the artifact file as str object
+
+    :param artifact_uri: artifact location
+    :return: str
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_text("This is a sentence", "file.txt")
+            file_content = mlflow.artifacts.load_text(artifact_uri + "/file.txt")
+            print(file_content)
+
+    .. code-block:: text
+        :caption: Output
+
+        This is a sentence
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return str(local_artifact_fd.read())
+            except Exception:
+                raise MlflowException("Unable to form a str object from file content", BAD_REQUEST)
+
+
+def load_dict(artifact_uri: str) -> dict:
+    """
+    Loads content of the artifact file as json/dict object
+
+    :param artifact_uri: artifact location
+    :return: dict
+
+    .. code-block:: python
+        :caption: Example
+
+        import mlflow
+        with mlflow.start_run() as run:
+            artifact_uri = run.info.artifact_uri
+            mlflow.log_dict({"mlflow-version": "0.28", "n_cores": "10"}, "config.json")
+            config_json = mlflow.artifacts.load_dict(artifact_uri + "/config.json")
+            print(config_json)
+
+    .. code-block:: text
+        :caption: Output
+
+        {'mlflow-version': '0.28', 'n_cores': '10'}
+    """
+    with tempfile.TemporaryDirectory() as tmpdir:
+        local_artifact = download_artifacts(artifact_uri, dst_path=tmpdir)
+        with open(local_artifact) as local_artifact_fd:
+            try:
+                return json.load(local_artifact_fd)
+            except json.JSONDecodeError:
+                raise MlflowException("Unable to form a JSON object from file content", BAD_REQUEST)
+
+
+def load_image(artifact_uri: str):
```

Comment:
```suggestion
def load_image(artifact_uri: str) -> 'PIL.Image':
```

@harupy  What is your suggestion on this change ?

Reply:
I think this is useless because Pillow is not typed:

https://pillow.readthedocs.io/en/stable/_modules/PIL/Image.html#open

Reply:
```python
# a.py

from PIL import Image

reveal_type(Image.open("foo"))
```

Output:

```
> mypy a.py
a.py:1: error: Skipping analyzing "PIL": module is installed, but missing library stubs or py.typed marker
a.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
a.py:3: note: Revealed type is "Any"
Found 1 error in 1 file (checked 1 source file)
```

Reply:
We need to do something like this to make type hint work:

```
from PIL import Image

import typing

if typing.TYPE_CHECKING:
    from PIL import Image


def get_image() -> "Image.Image":
    from PIL import Image

    return Image.open("test.png")


img = get_image()

```